llvm/lib/CodeGen/SelectionDAG/LegalizeIntegerTypes.cpp

```diff
@@ -2216,6 +2242,50 @@ SDValue DAGTypeLegalizer::PromoteIntOp_STORE(StoreSDNode *N, unsigned OpNo){
                            N->getMemoryVT(), N->getMemOperand());
 }
 
+SDValue DAGTypeLegalizer::PromoteIntOp_VP_STORE(VPStoreSDNode *N,
+                                                unsigned OpNo) {
+  SDValue DataOp = N->getValue();
+  SDValue Operand = N->getOperand(OpNo);
+
+  if (OpNo >= 4) {
+    // The Mask or EVL. Update in place.
```

Comment:
I suspect this is untested too.

Reply:
This is actually needed or vpstore.ll and fixed-vectors-vpstore.ll
```
$ llvm-lit test/CodeGen/RISCV/rvv/fixed-vectors-vpstore.ll -v
-- Testing: 1 tests, 1 workers --
FAIL: LLVM :: CodeGen/RISCV/rvv/fixed-vectors-vpstore.ll (1 of 1)
******************** TEST 'LLVM :: CodeGen/RISCV/rvv/fixed-vectors-vpstore.ll' FAILED ********************
Exit Code: 2

Command Output (stderr):
--
RUN: at line 2: /home/lei/llvm/community/build/bin/llc -mtriple=riscv32 -mattr=+d,+zvfh,+v    -verify-machineinstrs < /home/lei/llvm/community/llvm-project/llvm/test/CodeGen/RISCV/rvv/fixed-vectors-vpstore.ll | /home/lei/llvm/community/build/bin/FileCheck /home/lei/llvm/community/llvm-project/llvm/test/CodeGen/RISCV/rvv/fixed-vectors-vpstore.ll
+ /home/lei/llvm/community/build/bin/llc -mtriple=riscv32 -mattr=+d,+zvfh,+v -verify-machineinstrs
+ /home/lei/llvm/community/build/bin/FileCheck /home/lei/llvm/community/llvm-project/llvm/test/CodeGen/RISCV/rvv/fixed-vectors-vpstore.ll
PromoteIntegerOperand Op #1: t14: ch = vp_store<(store unknown-size into %ir.ptr, align 8)> t0, t12, t6, undef:i32, t9, t11

LLVM ERROR: Do not know how to promote this operator's operand!
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
```

Reply:
Isn't that failing on Operand 1? I was saying Operand 4 is untested.

Reply:
Sorry.  I thought you meant this function was also not needed.